### PR TITLE
WT-8259 Expose verbose categories and levels through WT_EVENT_HANDLER message strings

### PR DESCRIPTION
**Pull request by etienneptl**
We want to expose the verbose category and verbosity level as additional metadata in the message.

The key changes are:

- Update the internal '__eventv' method to take in a verbose category and level metadata fields.
- Support internal message paths (using the event handler interface) that don't specify verbose categories & levels - i.e. ‘wt_err’ & ‘wt_panic’ 
- Define a default verbose category these paths can fall back on.
- Default uses of these methods to the appropriate verbosity level i.e.  ‘WT_VERBOSE_ERROR’.

### CODE REVIEW

File: src/include/verbose.h [hunk 1]
```diff
@@ -64,11 +79,38 @@ typedef enum {
     WT_VERBOSE_DEBUG
 } WT_VERBOSE_LEVEL;
 
+/* Convert a verbose level to its string representation. */
+#define WT_VERBOSE_LEVEL_STR(level, level_str) \
+    do {                                       \
+        level_str = "";                        \
```
Comment (ddanderson) [inline src/include/verbose.h:85]: Since you're doing it elsewhere, parenthesize macro arg `level_str`

**Review by ddanderson — APPROVED** [review]
LGTM with a couple nits.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM!

File: src/include/verbose.h (hunk 1 above)
Comment (etienneptl) [reply src/include/verbose.h]: I should indeed! Fixed in 16c7aca2f41c8d211059fa909d2ed4dad669f623

File: src/include/verbose.h [hunk 2]
```diff
@@ -6,51 +6,66 @@
  * See the file LICENSE for redistribution information.
  */
 
+#define WT_GEN_VERBOSE_ENUM(enum) enum,
```
[the remark this answers is not in the document]
Comment (etienneptl) [reply src/include/verbose.h]: Renamed in def8a95d14846710a0e7c0223b3930f40e19a4cc

**Review by alisonfel — APPROVED** [review]
LGTM 👍

**Pull request merged**